hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -1997,6 +1999,10 @@ public QueryGroup visitQueryGroup(SqmQueryGroup<?> queryGroup) {
 				newQueryParts.add( visitQueryPart( queryParts.get( i ) ) );
 			}
 
+			// Register the table groups from the first sub-queryspec's FromClauseIndex
+			// as they might be needed to resolve the tuple type's table groups
+			firstQueryPartIndex.copyInto( fromClauseIndexStack.getCurrent() );
```

Comment:
Should be fine as the current from-clause-index (the query group's one) will no longer be used, unless in some very specific cases with `lastPoppedFromClauseIndex` (like here, where we have a `SqmDerivedRoot`).

Reply:
This is done after processing all query parts, as the very last operation before popping the stack, so it shouldn't affect their processing negatively AFAIU.

Overriding the `lastPoppedFromClauseIndex` and `lastPoppedProcessingState` does sound better though, and prevents modifying `FromClauseIndex` with an internal method I didn't really like.

Reply:
working on the changes